nanoFramework.Hardware.Esp32.Rmt/TransmitChannelSettings.cs

```diff
@@ -0,0 +1,138 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// <see cref="TransmitterChannel"/>'s settings class.
+    /// </summary>
+    /// <remarks>
+    /// All changes made to properties of this class are ignored after the <see cref="TransmitterChannel"/> is initialized.
+    /// The equivalent properties in the channel instance can be used to make on-the-fly changes to the configurations.
+    /// </remarks>
+    public sealed class TransmitChannelSettings : RmtChannelSettings
+    {
+        private bool _enableCarrierWave;
+        private bool _carrierLevel;
+        private int _carrierWaveFrequency;
+        private byte _carrierWaveDutyPercentage;
+        private bool _enableLooping;
+        private bool _enableIdleLevelOutput;
+        private bool _idleLevel;
+
+        /// <summary>
+        /// Enables or disables the carrier wave generator in the RMT Hardware.
+        /// </summary>
+        public bool EnableCarrierWave 
+        { 
+            get => _enableCarrierWave; 
+            set => _enableCarrierWave = value; 
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating at which level of RMT output is the carrier wave applied.
+        /// <see langword="true" /> = HIGH.
+        /// </summary>
+        public bool CarrierLevel 
+        { 
+            get => _carrierLevel; 
+            set => _carrierLevel = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the carrier wave frequency.
+        /// </summary>
+        public int CarrierWaveFrequency 
+        { 
+            get => _carrierWaveFrequency; 
+            set => _carrierWaveFrequency = value;
```

Comment:
similar question as for the rest, any range value?

Reply:
None, according to the IDF docs. In IDF, it takes in a `uint32_t`. I've added a check to make sure it is at least 1.